engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/SelectGameScreen.java

```diff
@@ -85,26 +88,43 @@ public String get() {
         refreshList(gameList);
         gameList.select(0);
         gameList.subscribe((widget, item) -> loadGame(item));
+        gameList.subscribeSelection((widget, selectedGame) -> {
```

Comment:
My first guess for the problem described by @mkienenb is that the `select(0)` happens too early, i.e., before the subscription is set up. Moving `gameList.select(0);` a few lines lower might already fix the problem, as `select(...)` will notifiy listeners (see https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/rendering/nui/widgets/UIList.java#L203-L204)